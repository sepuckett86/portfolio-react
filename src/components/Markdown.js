import React from 'react';
import styles from './Markdown.css';
import PropTypes from 'prop-types';
import ReactMarkdown from 'react-markdown';
import gfm from 'remark-gfm';

function Markdown({ content }) {
  return (
    <div className={styles.Markdown}>
      <div className={styles['markdown-body']}>
        <ReactMarkdown remarkPlugins={[gfm]}>
          {content}
        </ReactMarkdown>
      </div>
    </div>
  );
}

Markdown.propTypes = {
  content: PropTypes.string.isRequired,
};

export default Markdown;
