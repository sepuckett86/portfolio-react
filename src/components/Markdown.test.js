import React from 'react';
import { shallow } from 'enzyme';
import Markdown from './Markdown';

describe('Markdown component', () => {
  it('renders', () => {
    const content = 'markdown here';
    const wrapper = shallow(<Markdown content={content}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
