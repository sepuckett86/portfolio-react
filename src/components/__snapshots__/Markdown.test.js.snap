// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown component renders 1`] = `
<div
  className="Markdown"
>
  <div
    className="markdown-body"
  >
    <ReactMarkdown
      remarkPlugins={
        Array [
          [Function],
        ]
      }
      transformLinkUri={[Function]}
    >
      markdown here
    </ReactMarkdown>
  </div>
</div>
`;
